File: infix_to_postfix.py
def priority(op):
    if op == '^':
        return 3
    if op == '*' or op == '/' or op == '%':
        return 2
    if op == '+' or op == '-':
        return 1
    return 0

stack = []
postfix = ""

infix = input("Enter infix expression: ")

for ch in infix:

    if ch.isalnum():
        postfix += ch

    elif ch == '(':
        stack.append(ch)

    elif ch == ')':
        while stack and stack[-1] != '(':
            postfix += stack.pop()
        stack.pop()

    else:
        while stack and priority(stack[-1]) >= priority(ch):
            postfix += stack.pop()
        stack.append(ch)

while stack:
    postfix += stack.pop()

print("Postfix expression:", postfix)

File: infix_to_postfix.cpp
// This Program is to Convert Infix Expression to Postfix Expression using Stack Data Structure in C++
#include <iostream>
#include<string>
#include <stack>
using namespace std;

int priority(char op) {
    if (op == '^')
        return 3;
    if (op == '*' || op == '/' || op == '%')
        return 2;
    if (op == '+' || op == '-')
        return 1;
    return 0;
}

int main() {
    string ifix, pfix = "";
    stack<char> st;

    cout << "Enter infix expression: ";
    cin >> ifix;

    for (int i = 0; i < ifix.length(); i++) {
        char ch = ifix[i];

        if (isalnum(ch)) {
            pfix += ch;
        }
        
        else if (ch == '(') {
            st.push(ch);
        }
        
        else if (ch == ')') {
            while (!st.empty() && st.top() != '(') {
                pfix += st.top();
                st.pop();
            }
            st.pop();
        }
        
        else {
            while (!st.empty() && priority(st.top()) >= priority(ch)) {
                pfix += st.top();
                st.pop();
            }
            st.push(ch);
        }
    }

    while (!st.empty()) {
        pfix += st.top();
        st.pop();
    }

    cout << "Postfix expression: " << pfix << endl;

    return 0;
}
